circleci: failed tests: TestHeartbeatHealth

The following test appears to have failed:

[#19491](https://circleci.com/gh/cockroachdb/cockroach/19491):

```
--- SKIP: TestClockOffsetMetrics (0.01s)
	clock_offset_test.go:163: 
=== RUN   TestHeartbeatCB
--- PASS: TestHeartbeatCB (0.15s)
=== RUN   TestHeartbeatHealth
--- FAIL: TestHeartbeatHealth (15.27s)
	<autogenerated>:32: rpc/context_test.go:125: condition failed to evaluate within 15s: rpc/context_test.go:122: expected 127.0.0.1:58535 to be healthy
=== RUN   TestOffsetMeasurement
--- PASS: TestOffsetMeasurement (0.31s)
=== RUN   TestFailedOffsetMeasurement
--- PASS: TestFailedOffsetMeasurement (0.12s)
=== RUN   TestRemoteOffsetUnhealthy
I160622 21:44:14.246232 rpc/clock_offset.go:135  2 of 3 nodes are within the maximum offset of 100ms
I160622 21:44:14.246521 rpc/clock_offset.go:135  2 of 3 nodes are within the maximum offset of 100ms
I160622 21:44:14.246705 rpc/clock_offset.go:135  2 of 3 nodes are within the maximum offset of 100ms
I160622 21:44:14.252815 http2_client.go:919  transport: http2Client.notifyError got notified that the client transport was broken EOF.
--
--- PASS: TestRemoteOffsetString (0.01s)
=== RUN   TestHeartbeatReply
--- PASS: TestHeartbeatReply (0.02s)
=== RUN   TestManualHeartbeat
--- PASS: TestManualHeartbeat (0.01s)
FAIL
FAIL	github.com/cockroachdb/cockroach/rpc	17.252s
=== RUN   TestGetCertificateUser
--- PASS: TestGetCertificateUser (0.02s)
=== RUN   TestAuthenticationHook
--- PASS: TestAuthenticationHook (0.01s)
=== RUN   TestGenerateCerts
--- PASS: TestGenerateCerts (0.11s)
=== RUN   TestUseCerts
I160622 21:44:28.268027 storage/engine/rocksdb.go:143  opening in memory rocksdb instance
W160622 21:44:28.274284 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
W160622 21:44:28.283173 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node

```
Please assign, take a look and update the issue accordingly.

--- a/rpc/context_test.go
+++ b/rpc/context_test.go
@@ -109,39 +109,59 @@ func TestHeartbeatHealth(t *testing.T) {
 
 	clientCtx := newNodeTestContext(clock, stopper)
 	// Make the intervals and timeouts shorter to speed up the tests.
-	clientCtx.HeartbeatInterval = 10 * time.Millisecond
-	clientCtx.HeartbeatTimeout = 20 * time.Millisecond
+	clientCtx.HeartbeatInterval = 1 * time.Millisecond
+	clientCtx.HeartbeatTimeout = 1 * time.Millisecond
 	if _, err := clientCtx.GRPCDial(remoteAddr); err != nil {
 		t.Fatal(err)
 	}
 
+	// This code is inherently racy so when we need to verify heartbeats we want
+	// them to always succeed.
+	sendHeartbeats := func() func() {
+		done := make(chan struct{})
+		go func() {
+			for {
+				select {
+				case <-done:
+					return
+				case heartbeat.ready <- struct{}{}:
+				default:
+					time.Sleep(500 * time.Microsecond)
+				}
+			}
+		}()
+		return func() {
+			done <- struct{}{}
+		}
+	}
+
 	// Should be healthy after the first successful heartbeat.
-	heartbeat.ready <- struct{}{}
+	stopHeartbeats := sendHeartbeats()
 	util.SucceedsSoon(t, func() error {
 		if !clientCtx.IsConnHealthy(remoteAddr) {
 			return util.Errorf("expected %s to be healthy", remoteAddr)
 		}
 		return nil
 	})
+	stopHeartbeats()
 
-	// Should no longer be healthy after later heartbeats fail.
+	// Should no longer be healthy after heartbeating stops.
 	util.SucceedsSoon(t, func() error {
 		if clientCtx.IsConnHealthy(remoteAddr) {
 			return util.Errorf("expected %s to be unhealthy", remoteAddr)
 		}
 		return nil
 	})
-	// Complete the heartbeat that timed out on the server side.
-	heartbeat.ready <- struct{}{}
 
 	// Should return to healthy after another successful heartbeat.
-	heartbeat.ready <- struct{}{}
+	stopHeartbeats = sendHeartbeats()
 	util.SucceedsSoon(t, func() error {
 		if !clientCtx.IsConnHealthy(remoteAddr) {
 			return util.Errorf("expected %s to be healthy", remoteAddr)
 		}
 		return nil
 	})
+	stopHeartbeats()
 
 	if clientCtx.IsConnHealthy("non-existent connection") {
 		t.Errorf("non-existent connection is reported as healthy")

